src/s3/client.rs

```diff
@@ -203,6 +204,91 @@ fn parse_list_objects_common_prefixes(
     Ok(())
 }
 
+/// Client Builder manufactures a Client using given parameters.
+#[derive(Debug, Default)]
+pub struct ClientBuilder {
+    base_url: BaseUrl,
+    provider: Option<Arc<Box<(dyn Provider + Send + Sync + 'static)>>>,
+    ssl_cert_file: Option<PathBuf>,
+    ignore_cert_check: Option<bool>,
+    user_agent: Option<String>,
+}
+
+impl ClientBuilder {
+    /// Creates a builder given a base URL for the MinIO service or other AWS S3
+    /// compatible object storage service.
+    pub fn new(base_url: BaseUrl) -> Self {
+        let mut c = ClientBuilder::default();
+        c.base_url = base_url;
+        c
+    }
+
+    /// Set the credential provider. If not set anonymous access is used.
+    pub fn provider(
+        mut self,
+        provider: Option<Box<(dyn Provider + Send + Sync + 'static)>>,
+    ) -> Self {
+        self.provider = provider.map(Arc::new);
+        self
+    }
+
+    /// Set the user agent.
+    pub fn user_agent(mut self, user_agent: Option<String>) -> Self {
+        self.user_agent = user_agent;
+        self
+    }
+
+    /// Set file for loading a trust certificate.
+    pub fn ssl_cert_file(mut self, ssl_cert_file: Option<&Path>) -> Self {
+        self.ssl_cert_file = ssl_cert_file.map(PathBuf::from);
+        self
+    }
+
+    /// Set the ignore_cert_check.
```

Comment:
You could have `/// Set flag to ignore certificate check`

Reply:
Done